Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -22,47 +22,75 @@ public IonObjectSerializer(IonSerializer ionSerializer, IonSerializationOptions
 
         public object Deserialize(IIonReader reader)
         {
-            var targetObject = options.ObjectFactory.Create(options, reader, targetType);
-            reader.StepIn();
-
-            IonType ionType;
-            while ((ionType = reader.MoveNext()) != IonType.None)
+            if (reader == null)
+            {
+                throw new InvalidOperationException("Cannot deserialize with a null Ion reader");
```

Comment:
Seems like this should be https://docs.microsoft.com/en-us/dotnet/api/system.argumentexception?view=net-5.0 instead.

Reply:
Addressed.